File: crossecore-java/src/java_/IfStatementBase.java
package java_;
import org.eclipse.emf.common.notify.Notification;
import org.eclipse.emf.common.notify.NotificationChain;
import org.eclipse.emf.common.notify.impl.NotificationImpl;
import org.eclipse.emf.ecore.InternalEObject;
import org.eclipse.emf.ecore.*;
import org.eclipse.emf.ecore.impl.ENotificationImpl;
public class IfStatementBase 
extends StatementImpl implements IfStatement
{
	private Expression _expression;
	public Expression getExpression()
	{
		
			return _expression;
		
		
	}
	public void setExpression(Expression value){
		if (value != _expression) {
			NotificationChain msgs = null;
			if (_expression != null){
				msgs = ((InternalEObject)_expression).eInverseRemove(this, EOPPOSITE_FEATURE_BASE - Java_PackageImpl.IF_STATEMENT__EXPRESSION, null, msgs);
			}
			if (value != null){
				msgs = ((InternalEObject)value).eInverseAdd(this, EOPPOSITE_FEATURE_BASE - Java_PackageImpl.IF_STATEMENT__EXPRESSION, null, msgs);
			}
			msgs = basicSetExpression(value, msgs);
			if (msgs != null) {
				msgs.dispatch();
			}
		}
		else if (eNotificationRequired()){
			eNotify(new ENotificationImpl(this, NotificationImpl.SET,Java_PackageImpl.IF_STATEMENT__EXPRESSION , value, value));
		}
	}
	private Statement _thenStatement;
	public Statement getThenStatement()
	{
		
			return _thenStatement;
		
		
	}
	public void setThenStatement(Statement value){
		if (value != _thenStatement) {
			NotificationChain msgs = null;
			if (_thenStatement != null){
				msgs = ((InternalEObject)_thenStatement).eInverseRemove(this, EOPPOSITE_FEATURE_BASE - Java_PackageImpl.IF_STATEMENT__THEN_STATEMENT, null, msgs);
			}
			if (value != null){
				msgs = ((InternalEObject)value).eInverseAdd(this, EOPPOSITE_FEATURE_BASE - Java_PackageImpl.IF_STATEMENT__THEN_STATEMENT, null, msgs);
			}
			msgs = basicSetThenStatement(value, msgs);
			if (msgs != null) {
				msgs.dispatch();
			}
		}
		else if (eNotificationRequired()){
			eNotify(new ENotificationImpl(this, NotificationImpl.SET,Java_PackageImpl.IF_STATEMENT__THEN_STATEMENT , value, value));
		}
	}
	private Statement _elseStatement;
	public Statement getElseStatement()
	{
		
			return _elseStatement;
		
		
	}
	public void setElseStatement(Statement value){
		if (value != _elseStatement) {
			NotificationChain msgs = null;
			if (_elseStatement != null){
				msgs = ((InternalEObject)_elseStatement).eInverseRemove(this, EOPPOSITE_FEATURE_BASE - Java_PackageImpl.IF_STATEMENT__ELSE_STATEMENT, null, msgs);
			}
			if (value != null){
				msgs = ((InternalEObject)value).eInverseAdd(this, EOPPOSITE_FEATURE_BASE - Java_PackageImpl.IF_STATEMENT__ELSE_STATEMENT, null, msgs);
			}
			msgs = basicSetElseStatement(value, msgs);
			if (msgs != null) {
				msgs.dispatch();
			}
		}
		else if (eNotificationRequired()){
			eNotify(new ENotificationImpl(this, NotificationImpl.SET,Java_PackageImpl.IF_STATEMENT__ELSE_STATEMENT , value, value));
		}
	}

	
	@Override
	protected EClass eStaticClass() {
		return Java_PackageImpl.Literals.IFSTATEMENT;
	}
	
	
	public NotificationChain basicSetThenStatement(Statement newobj, NotificationChain msgs) {
		Statement oldobj = _thenStatement;
		_thenStatement = newobj;
		if (eNotificationRequired()) {
			ENotificationImpl notification = new ENotificationImpl(this, NotificationImpl.SET, Java_PackageImpl.IF_STATEMENT__THEN_STATEMENT, oldobj, newobj);
			if (msgs == null){
				msgs = notification;
			}
			else{
				msgs.add(notification);
			}
		}
		return msgs;
	}
	public NotificationChain basicSetExpression(Expression newobj, NotificationChain msgs) {
		Expression oldobj = _expression;
		_expression = newobj;
		if (eNotificationRequired()) {
			ENotificationImpl notification = new ENotificationImpl(this, NotificationImpl.SET, Java_PackageImpl.IF_STATEMENT__EXPRESSION, oldobj, newobj);
			if (msgs == null){
				msgs = notification;
			}
			else{
				msgs.add(notification);
			}
		}
		return msgs;
	}
	public NotificationChain basicSetElseStatement(Statement newobj, NotificationChain msgs) {
		Statement oldobj = _elseStatement;
		_elseStatement = newobj;
		if (eNotificationRequired()) {
			ENotificationImpl notification = new ENotificationImpl(this, NotificationImpl.SET, Java_PackageImpl.IF_STATEMENT__ELSE_STATEMENT, oldobj, newobj);
			if (msgs == null){
				msgs = notification;
			}
			else{
				msgs.add(notification);
			}
		}
		return msgs;
	}
	
	@Override
	public Object eGet(int featureID, boolean resolve, boolean coreType) {
		switch (featureID) {
			case Java_PackageImpl.IF_STATEMENT__EXPRESSION:
				return getExpression();
			case Java_PackageImpl.IF_STATEMENT__THEN_STATEMENT:
				return getThenStatement();
			case Java_PackageImpl.IF_STATEMENT__ELSE_STATEMENT:
				return getElseStatement();
		}
		return super.eGet(featureID, resolve, coreType);
	}
	
	@Override
	public void eSet(int featureID, Object newValue) {
		switch (featureID) {
			case Java_PackageImpl.IF_STATEMENT__EXPRESSION:
				setExpression((Expression) newValue);
				return;
			case Java_PackageImpl.IF_STATEMENT__THEN_STATEMENT:
				setThenStatement((Statement) newValue);
				return;
			case Java_PackageImpl.IF_STATEMENT__ELSE_STATEMENT:
				setElseStatement((Statement) newValue);
				return;
		}
		super.eSet(featureID, newValue);
	}
	
	@Override
	public boolean eIsSet(int featureID) {
		switch (featureID) {
			case Java_PackageImpl.IF_STATEMENT__EXPRESSION:
				return _expression != null; //single != null;
			case Java_PackageImpl.IF_STATEMENT__THEN_STATEMENT:
				return _thenStatement != null; //single != null;
			case Java_PackageImpl.IF_STATEMENT__ELSE_STATEMENT:
				return _elseStatement != null; //single != null;
		}
		return super.eIsSet(featureID);
	}
	
	
	
}

File: crossecore-csharp/Java/Java/Java_/IfStatementBase.cs
/* CrossEcore is a cross-platform modeling framework that generates C#, TypeScript, 
 * JavaScript, Swift code from Ecore models with embedded OCL (http://www.crossecore.org/).
 * The original Eclipse Modeling Framework is available at https://www.eclipse.org/modeling/emf/.
 * 
 * contributor: Simon Schwichtenberg
 */
 
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using oclstdlib;
using Ecore;
namespace Java_{
	public class IfStatementBase 
	:StatementImpl, IfStatement
	{
		private Expression _expression;
		public virtual Expression expression
		{
			get {
			
				return _expression;
			}
			set {
				if (value != _expression) {
					NotificationChain msgs = null;
					if (_expression != null){
						msgs = ((InternalEObject)_expression).eInverseRemove(this, EOPPOSITE_FEATURE_BASE - Java_PackageImpl.IFSTATEMENT_EXPRESSION, null, msgs);
					}
					if (value != null){
						msgs = ((InternalEObject)value).eInverseAdd(this, EOPPOSITE_FEATURE_BASE - Java_PackageImpl.IFSTATEMENT_EXPRESSION, null, msgs);
					}
					msgs = basicSetExpression(value, msgs);
					if (msgs != null) {
						msgs.dispatch();
					}
				}
				else if (eNotificationRequired()){
					eNotify(new ENotificationImpl(this, NotificationImpl.SET,Java_PackageImpl.IFSTATEMENT_EXPRESSION , value, value));
				}
				}
		}
		private Statement _thenStatement;
		public virtual Statement thenStatement
		{
			get {
			
				return _thenStatement;
			}
			set {
				if (value != _thenStatement) {
					NotificationChain msgs = null;
					if (_thenStatement != null){
						msgs = ((InternalEObject)_thenStatement).eInverseRemove(this, EOPPOSITE_FEATURE_BASE - Java_PackageImpl.IFSTATEMENT_THENSTATEMENT, null, msgs);
					}
					if (value != null){
						msgs = ((InternalEObject)value).eInverseAdd(this, EOPPOSITE_FEATURE_BASE - Java_PackageImpl.IFSTATEMENT_THENSTATEMENT, null, msgs);
					}
					msgs = basicSetThenStatement(value, msgs);
					if (msgs != null) {
						msgs.dispatch();
					}
				}
				else if (eNotificationRequired()){
					eNotify(new ENotificationImpl(this, NotificationImpl.SET,Java_PackageImpl.IFSTATEMENT_THENSTATEMENT , value, value));
				}
				}
		}
		private Statement _elseStatement;
		public virtual Statement elseStatement
		{
			get {
			
				return _elseStatement;
			}
			set {
				if (value != _elseStatement) {
					NotificationChain msgs = null;
					if (_elseStatement != null){
						msgs = ((InternalEObject)_elseStatement).eInverseRemove(this, EOPPOSITE_FEATURE_BASE - Java_PackageImpl.IFSTATEMENT_ELSESTATEMENT, null, msgs);
					}
					if (value != null){
						msgs = ((InternalEObject)value).eInverseAdd(this, EOPPOSITE_FEATURE_BASE - Java_PackageImpl.IFSTATEMENT_ELSESTATEMENT, null, msgs);
					}
					msgs = basicSetElseStatement(value, msgs);
					if (msgs != null) {
						msgs.dispatch();
					}
				}
				else if (eNotificationRequired()){
					eNotify(new ENotificationImpl(this, NotificationImpl.SET,Java_PackageImpl.IFSTATEMENT_ELSESTATEMENT , value, value));
				}
				}
		}
		
		protected override EClass eStaticClass() {
			return Java_PackageImpl.Literals.IFSTATEMENT;
		}
		
		
		public NotificationChain basicSetExpression(Expression newobj, NotificationChain msgs) {
			var oldobj = _expression;
			_expression = newobj;
			if (eNotificationRequired()) {
				var notification = new ENotificationImpl(this, NotificationImpl.SET, Java_PackageImpl.IFSTATEMENT_EXPRESSION, oldobj, newobj);
				if (msgs == null){
					msgs = notification;
				}
				else{
					msgs.add(notification);
				}
			}
			return msgs;
		}
		public NotificationChain basicSetThenStatement(Statement newobj, NotificationChain msgs) {
			var oldobj = _thenStatement;
			_thenStatement = newobj;
			if (eNotificationRequired()) {
				var notification = new ENotificationImpl(this, NotificationImpl.SET, Java_PackageImpl.IFSTATEMENT_THENSTATEMENT, oldobj, newobj);
				if (msgs == null){
					msgs = notification;
				}
				else{
					msgs.add(notification);
				}
			}
			return msgs;
		}
		public NotificationChain basicSetElseStatement(Statement newobj, NotificationChain msgs) {
			var oldobj = _elseStatement;
			_elseStatement = newobj;
			if (eNotificationRequired()) {
				var notification = new ENotificationImpl(this, NotificationImpl.SET, Java_PackageImpl.IFSTATEMENT_ELSESTATEMENT, oldobj, newobj);
				if (msgs == null){
					msgs = notification;
				}
				else{
					msgs.add(notification);
				}
			}
			return msgs;
		}
		
		public override object eGet(int featureID, bool resolve, bool coreType) {
			switch (featureID) {
				case Java_PackageImpl.IFSTATEMENT_EXPRESSION:
					return expression;
				case Java_PackageImpl.IFSTATEMENT_THENSTATEMENT:
					return thenStatement;
				case Java_PackageImpl.IFSTATEMENT_ELSESTATEMENT:
					return elseStatement;
			}
			return base.eGet(featureID, resolve, coreType);
		}
		
		
		public override void eSet(int featureID, object newValue) {
			switch (featureID) {
				case Java_PackageImpl.IFSTATEMENT_EXPRESSION:
					expression = (Expression) newValue;
					return;
				case Java_PackageImpl.IFSTATEMENT_THENSTATEMENT:
					thenStatement = (Statement) newValue;
					return;
				case Java_PackageImpl.IFSTATEMENT_ELSESTATEMENT:
					elseStatement = (Statement) newValue;
					return;
			}
			base.eSet(featureID, newValue);
		}
		
		/*
		public override bool eIsSet(int featureID) {
			switch (featureID) {
				case Java_PackageImpl.IFSTATEMENT_EXPRESSION:
					return _expression != null; //single != null;
				case Java_PackageImpl.IFSTATEMENT_THENSTATEMENT:
					return _thenStatement != null; //single != null;
				case Java_PackageImpl.IFSTATEMENT_ELSESTATEMENT:
					return _elseStatement != null; //single != null;
			}
			return base.eIsSet(featureID);
		}
		*/
		
		
		
	}
}
